crates/red_knot_python_semantic/src/types.rs

```diff
@@ -824,6 +825,54 @@ impl<'db> From<&Type<'db>> for Type<'db> {
     }
 }
 
+/// Non-exhaustive enumeration of builtin types to allow for easier syntax when interacting with
+/// the most common builtin types (e.g. int, str, ...).
+///
+/// Feel free to expend this enum if you ever find yourself using the same builtin type in multiple
+/// places.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, strum_macros::EnumIter)]
+pub enum BuiltinType {
+    Bool,
+    Object,
+    Bytes,
+    Int,
+    Float,
+    Str,
+    List,
+    Tuple,
+    Set,
+    Dict,
+}
+
+impl<'db> BuiltinType {
+    pub const fn as_str(&self) -> &'static str {
+        match self {
+            Self::Bool => "bool",
+            Self::Object => "object",
+            Self::Bytes => "bytes",
+            Self::Tuple => "tuple",
+            Self::Int => "int",
+            Self::Float => "float",
+            Self::Str => "str",
+            Self::Set => "set",
+            Self::Dict => "dict",
+            Self::List => "list",
+        }
+    }
+
+    pub fn to_instance(&self, db: &'db dyn Db) -> Type<'db> {
+        builtins_symbol_ty(db, self.as_str()).to_instance(db)
+    }
+
+    pub fn to_class(&self, db: &'db dyn Db) -> Type<'db> {
+        builtins_symbol_ty(db, self.as_str())
+    }
+
+    pub fn from_name(name: &str) -> Option<Self> {
+        BuiltinType::iter().find(|&builtin| builtin.as_str() == name)
+    }
```

Comment:
What's the reason for introducing and using `BuiltinType::iter` here instead of a match. I think I'd prefer to avoid adding a dependency for this use case as it feels a bit overkill :)

Reply:
I guess the reason was to make it easier to add more members as needed - but that's probably not worth a dependency!